app/main.py

```diff
@@ -12,39 +12,17 @@ def format_linter_error(error: dict) -> dict:
 def format_single_linter_file(file_path: str, errors: list) -> dict:
 
     return {
-        "errors": [
-            {
-                "line": error["line_number"],
-                "column": error["column_number"],
-                "message": error["text"],
-                "name": error["code"],
-                "source": "flake8"
-            } for error in errors
-        ],
+        "errors": [format_linter_error(error) for error in errors],
         "path": file_path,
-        "status": "failed"
+        "status": "failed" if errors else "passed"
     }
 
 
 def format_linter_report(linter_report: dict) -> list:
-
     return [
         {
-            "errors": report,
-            "path": path,
-            "status": "passed"
-        } if not report
-        else {
-            "errors": [
-                {
-                    "line": error["line_number"],
-                    "column": error["column_number"],
-                    "message": error["text"],
-                    "name": error["code"],
-                    "source": "flake8"
-                } for error in report if report
-            ],
+            "errors": format_single_linter_file(path, report)["errors"],
             "path": path,
-            "status": "failed"
+            "status": "passed" if not report else "failed"
```

Comment:
You have path and status already implemented in format_single_linter_file. Think about handling reports in format_linter_report 1 by 1 using format_single_linter_file

Reply:
Do you mean I need to use in my third function params from my second function? And I should replace path -> file_path and report -> errors like here?
{
24 "errors": format_single_linter_file(file_path, errors)["errors"],
25 "path": file_path,
26 "status": "passed" if not errors else "failed"
27  for file_path, errors in linter_report.items()
}
Or my third function should accept data from my second function and it should define "path" and "status" by itself without any extra conditions like in this line 
26 "status": "passed" if not errors else "failed"
where I repeat the same action as in the second function.

Reply:
And is it good practice to handle conversation here?